pangeo_forge_recipes/transforms.py

```diff
@@ -51,7 +51,11 @@ class OpenWithXarray(beam.PTransform):
 
     def _open_with_xarray(self, element: Tuple[Index, Any]) -> Tuple[Index, xr.Dataset]:
         key, open_file = element
-        ds = xr.open_dataset(open_file, **self.xarray_open_kwargs)
+        try:
+            ds = xr.open_dataset(open_file, **self.xarray_open_kwargs)
+        except ValueError:
+            with open_file as fp:
+                ds = xr.open_dataset(fp, **self.xarray_open_kwargs)
```

Comment:
If I try to just always use the context manager, like this,
```suggestion
        with open_file as fp:
            ds = xr.open_dataset(fp, **self.xarray_open_kwargs)
```

I get the error

```
E   TypeError: cannot pickle '_io.BufferedReader' object
[while running 'Create|OpenWithFSSpec|OpenWithXarray/OpenWithXarray/Open with Xarray']
```

This only happens when opening from the cache. In that case, we are not using `fsspec.open` but rather using the filesystem `open` method that is attached to the cache object, i.e. 

https://github.com/rabernat/pangeo-forge-recipes/blob/0a15ccfeb2fbcf14ac569e4dac86a9b99329073a/pangeo_forge_recipes/storage.py#L129-L133

So we need to try to reconcile the behavior of `get_opener` with `cache.open_file`, such that they return the same type of open file object.